cli/internal/helm/actionfactory.go

```diff
@@ -71,13 +69,21 @@ func (a actionFactory) GetActions(releases []Release, force, allowDestructive bo
 	return actions, includesUpgrade, errors.Join(upgradeErrs...)
 }
 
-func (a actionFactory) appendNewAction(release Release, force, allowDestructive bool, actions *[]applyAction) error {
+func (a actionFactory) appendNewAction(release Release, configTargetVersion semver.Semver, force, allowDestructive bool, actions *[]applyAction) error {
 	newVersion, err := semver.New(release.Chart.Metadata.Version)
 	if err != nil {
 		return fmt.Errorf("parsing chart version: %w", err)
 	}
 	currentVersion, err := a.versionLister.currentVersion(release.ReleaseName)
 	if errors.Is(err, errReleaseNotFound) {
+		// Don't install a new release if the user's config specifies a different version than the CLI offers.
+		if !force && isCLIVersionedRelease(release.ReleaseName) && configTargetVersion.Compare(newVersion) != 0 {
+			return fmt.Errorf(
+				"unable to install release %s at %s: this CLI only supports microservice version %s for upgrading",
```

Comment:
I think we should throw an InvalidUpgradeError in all error cases because we don't want to fail the entire upgrade but just skip right? Our policy was to skip on errors and not to fail entirely? Could we add an unit test that checks the error behavior for an invalid microservice version?

Reply:
We [previously returned a hard error if this was the case](https://github.com/edgelesssys/constellation/blob/094e0744d9aebd29469699d88ddb396b9246b54d/cli/internal/helm/upgrade.go#L106-L108).
But I'm open to changing this. Any opinions on this @derpsteb

Reply:
Before the helm refactoring we only showed the user that this upgrade won't be executed for $ERROR reasons and continued with other upgrades. I prefer that version over exiting after an error. 

As long as we don't mess up our version compatibility it should always be fine to miss one microservice upgrade while images are already upgraded.^^

Reply:
Ok, then lets just skip the upgrade with `InvalidUpgradeError` if the CLI does not support the version.